compiler/lib/generate.ml

```diff
@@ -1508,6 +1514,13 @@ and compile_block_no_loop st queue (pc : Addr.t) loop_stack frontier interm =
       let try_catch_frontier = Addr.Set.union new_frontier handler_frontier_cont in
       if debug () then Format.eprintf "@[<hv 2>try {@;";
       if Interm.mem pc1 handler_interm then decr_preds st pc1;
+      (* Parameters of the exception handler are assigned here, not in
+         handler body. Indeed, they correspond to variables that are
+         going to be modified by an instruction Assign. *)
+      let _, assign_handler_variables =
```

Comment:
Alternatively, we could rewrite the code, adding an intermediate branch before the `Pushtrap` than binds the exception handler's variable. Maybe as soon as in `Parse_bytecode`? Then, we do not have to do anything special here.

I'll have to do this transformation to handle effect handlers anyway.